test/Feature/HLSLLib/distance.32.denormal.test

```diff
@@ -0,0 +1,92 @@
+#--- source.hlsl
+
+StructuredBuffer<float4> X : register(t0);
+StructuredBuffer<float4> Y : register(t1);
+
+RWStructuredBuffer<float> Result : register(u2);
+
+[numthreads(1,1,1)]
+void main() {
+  // test special cases
+  // distance(inf, 1.0) = inf
+  Result[0] = distance(X[0].x, Y[0].x);

```

Comment:
The degenerate scalar case is not that interesting for distance, since it just translates to fsub. Vector distance translates to fsub, fmul+fadd per component, sqrt.

So these scalar cases are really only testing the behavior of fsub.